src/arrayMethodSort.js

```diff
@@ -1,12 +1,63 @@
+/* eslint-disable no-param-reassign */
 'use strict';
 
 /**
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (compareFunction === undefined) {
+      compareFunction = (a, b) => {
+        const isStringA = typeof a === 'string';
+        const isStringB = typeof b === 'string';
+
+        if (isStringA && isStringB) {
+          const isUpperA = a.charCodeAt(0) >= 65 && a.charCodeAt(0) <= 90;
+          const isUpperB = b.charCodeAt(0) >= 65 && b.charCodeAt(0) <= 90;
+
+          if (isUpperA && !isUpperB) {
+            return -1;
+          }
+
+          if (!isUpperA && isUpperB) {
+            return 1;
+          }
+        }
+
+        return String(a).localeCompare(String(b));
+      };
+    }
+
+    quickSort(this, 0, this.length - 1, compareFunction);
```

Comment:
The `quickSort` function is used here, which is a more advanced and efficient sorting algorithm compared to a simple bubble sort. However, as per the task description, it is okay to use a simple bubble sort algorithm. If this is an intentional design decision for performance reasons, it's acceptable, but otherwise, consider using a simpler algorithm if the task requirements allow.